action/install.go

```diff
@@ -13,18 +13,24 @@ import (
 	"github.com/helm/helm/log"
 )
 
+// The order in which manifests should be installed, by Kind.
+//
+// Anything not on the list will be installed after the last listed item, in
+// an indeterminate order.
+var InstallOrder = []string{"Namespace", "Secret", "PersistentVolume", "ServiceAccount", "Service", "Pod", "ReplicationController", "DaemonSet", "Ingress", "Job"}
+
+// The order in which manifests are uninstalled.
+//
+// Unknown manifest types (those not explicitly referenced in this list) will
+// be uninstalled before any of these, since we know that none of the core
+// types depend on non-core types.
+var UninstallOrder = []string{"Service", "Pod", "ReplicationController", "DaemonSet", "Secret", "PersistentVolume", "ServiceAccount", "Ingress", "Job"}
```

Comment:
@bacongobbler that was a good catch. I had forgotten that if the type isn't explicitly set in this list, it gets deleted *before* all the declared ones. That would have had some nasty side-effects.